docs/NETWORK-ACLS.md

```diff
@@ -0,0 +1,289 @@
+# Network ACLs
+
+An access control list (ACL) consists of one or more access control entries (ACEs) that collectively define
+the network traffic profile.
+This document describes different types of ACEs, their semantics, limitations and provides few basic examples
+for common use-cases.
+
+## Configuration Model
+
+In EVE, ACLs are configured separately for every ECO interface.
+Protobuf-modeled configuration model of ACLs can be found in the [fw.proto file](../api/proto/config/fw.proto).
+To control the network traffic between a given ECO and other endpoints within the scope of a given network instance,
+add one or more ACEs under the `acls` field of [NetworkAdapter](../api/proto/config/netconfig.proto).
+
+Every ACE should have a unique non-zero integer ID assigned. Please note that ACE ID should not exceed 24bits.
+ACE ID is set in [flowlogs](../api/proto/flowlog/flowlog.proto) as `aclId` for every flow to show which ACE was applied.
+ACE `ID=0` is reserved for a default reject-all rule that implicitly exists at the end of every ACL (even flows whose
+packets are rejected are logged).
+
+## ACL Semantics
+
+Every ECO interface is assigned a single separate access control list filtering inbound and outbound traffic
+(unless it is a passthrough interface, in which case EVE networking is completely bypassed).
+A single entry of ACL, denoted as ACE, consists of a set of matches that use layer 3 and 4 information to select
+a subset of the traffic, and a set of actions deciding what to do with the selected traffic. Optionally, it is possible
+to limit ACE to only ingress or egress flow direction. By default, ACEs are bi-directional, i.e. applied irrespective
```

Comment:
I don't think we've ever testing setting direction. Subce we create connection state in conntrack the direction might (I don't know) just influence the direction needed to create the state.

Reply:
Here I was describing the specification (the API), which allows to set the direction.
But then in the limitations I mention that uni-directionality is actually not implemented and the option is prepared for the future.

Reply:
Maybe having a subtitle at the top saying it is about the API and semantics would make that more clear.

But also (and I'm fully to blame) the intended semantics are not clear because they are determined by what iptables+conntrack would actually do if we set some direction.

Reply:
The section title is "ACL Semantics".
Currently the direction is completely ignored by zedrouter and rules are installed as bidirectional (mentioned under "Limitations").
So we still have the freedom here to define what we want the semantics for unidirectional ACLs to be and implement it later (if possible with iptables+conntract).